### PR TITLE
Minor optimization to take_slice

### PR DESCRIPTION
**Pull request by shssoichiro**

### CODE REVIEW

File: src/rdo.rs [hunk 1]
```diff
@@ -478,11 +478,7 @@ pub fn spatiotemporal_scale<T: Pixel>(
   // compared to using `Iterator::take`
   #[inline(always)]
   fn take_slice<T>(slice: &[T], n: usize) -> &[T] {
-    if n >= slice.len() {
-      slice
-    } else {
-      &slice[..n]
-    }
+    slice.get(..n).unwrap_or(slice)
```
Comment (tdaede) [inline src/rdo.rs:481]: Might be worth documenting the new possible panic (I think if frame_bo is too close to the edge)

Comment (maj160) [reply src/rdo.rs:481]: I don't think this panics - if the get fails then the `unwrap_or(...)` (not `unwrap()`!) returns the original slice.

Comment (shssoichiro) [reply src/rdo.rs:481]: The behavior should be identical. `slice.get` returns an `Option` which is `None` if the requested index is out of range.

Comment (tdaede) [reply src/rdo.rs:481]: Indeed I misread that.

**Review by tdaede — APPROVED** [review]

**Pull request merged**